### ISSUE TITLE
ObjectiveDropboxOfficial cannot authorize app when dropbox app is installed in same device

### ISSUE THREAD

**Issue by Ranjan911** [dropbox/dropbox-sdk-obj-c#195]
I integrated ObjectiveDropboxOfficial SDK of dropbox into my app. I followed the dropbox tutorial provided and was able to authorize the app and do further dropbox operations like upload and download.
It worked well.

But if the same app is installed in a device which already has dropbox app installed, then the app is unable to get authorize from dropbox.
When my app call following code:
[DBClientsManager authorizeFromController:[UIApplication sharedApplication]
                                 controller:[[self class] topMostController]
                                    openURL:^(NSURL *url) {
                                      [[UIApplication sharedApplication] openURL:url];
                                    }];

The app goes to background and dropbox app gets open asking to allow access for dropbox. Here is the screenshot
![img_0124](https://user-images.githubusercontent.com/3942087/32362729-8932a86a-c091-11e7-8666-4e3d70e762d6.PNG)
When i tap allow, it goes back to my app and nothing happens. So, i again try to access dropbox in my app, it again goes to dropbox app and again ask for authorization. So, my app never gets authorized with dropbox.

Is there anyone, who faced similar problem while integrating ObjectiveDropboxOfficial SDK? Is there any cure for this?

**Comment by greg-db** [comment]
@Ranjan911 Can you double check that you configured your plist as shown here?

https://github.com/dropbox/dropbox-sdk-obj-c#application-plist-file

**Comment by Ranjan911** [comment]
@greg-db Yes, its all fine, because, dropbox works well when there is no dropbox app installed in my device.
It fails to authorize when there is dropbox app installed in device.

**Comment by greg-db** [comment]
For reference then, can you try configuring and running [the example app](https://github.com/dropbox/dropbox-sdk-obj-c/tree/master/Examples/DBRoulette), to see if the same issue appears there? That should help narrow down where the issue is. Thanks in advance!

**Comment by scobbe** [comment]
@Ranjan911: In addition to Greg's request, can you try running this command:
```
/System/Library/Frameworks/CoreServices.framework/Frameworks/LaunchServices.framework/Support/lsregister -kill -r -domain local -domain system -domain user
```

Sometimes if you have another build with the same app key, the redirect gets confused.
